File: modules/angular2/src/facade/js_interop.ts
import {global} from './lang';

export function setGlobalVar(name: string, value: any) {
  global[name] = value;
}

export function getGlobalVar(name: string) {
  return global[name];
}

export function invokeJsFunction(fn: Function, self: any, args: List<any>) {
  return fn.apply(self, args);
}

File: modules/angular2/src/facade/js_interop.dart
library angular2.src.facade.js_interop;

import 'dart:js' as js;

setGlobalVar(String name, value) {
  js.context[name] = value;
}

getGlobalVar(String name) {
  return js.context[name];
}

invokeJsFunction(js.JsFunction fn, self, args) {
  return fn.apply(args, thisArg: self);
}
